addons/binding/org.openhab.binding.lgwebos/README.md

```diff
@@ -0,0 +1,190 @@
+
+## Thing Configuration
+
+WebOS TV has no configuration parameters. Please note that at least one channel must be bound to an item before the binding will make an attempt to connect and pair with the TV once that one is turned on.
+
+## Channels
+
+| Channel Type ID | Item Type    | Description  | Read/Write |
+| --------------- | ------------ | ------------ | ---------- |
+| power | Switch | Current power setting. TV can only be powered off, not on.  | RW |
+| mute | Switch | Current mute setting.  |  RW |
+| volume | Dimmer | Current volume setting. Setting and reporting absolute percent values only works when using internal speakers. Connected to an external amp the volume should be controlled using increase and decrease relative commands. |  RW |
+| channel | String | Current channel | RW | 
+| channelUp | Switch | One channel up |  W |
+| channelDown | Switch | One channel down  |  W |
+| channelName | String | Current channel name |  R |
+| toast | String | Displays a short message on the TV screen. See also rules section. |  W |
+| mediaPlayer | Player | Media control player |  W |
+| mediaStop | Switch | Media control stop  |  W |
+| appLauncher | String | Application ID of currently running application. This also allows to start applications on the TV by sending a specific Application ID to this channel. |  RW |
+
+
+## Actions
+
+### Show Toast
+
+```
+showToast(String ip, String text)
+```
+
+Sends a toast message to a webOS device using OpenHAB's logo as an icon.
+The first parameter is the IP address of your TV. 
+The second parameter is the message you want to display.
+### Show Toast with Custom Icon
+
+```
+showToast(String ip, String icon, String text)
+```
+
+Sends a toast message to a webOS device with custom icon. 
+The first parameter is the IP address of your TV. 
+The second parameter for the icon has to be provided as a URL. To use openhab's icon set you could send this URL for example: http://localhost:8080/icon/energy?state=UNDEF&format=png
+The third parameter is the message you want to display.
+
+### Launch a URL
+
+```
+launchBrowser(String ip, String url)
+```
+
+Opens the given URL in the TV's browser app.
+
+The first parameter is the IP address of your TV. 
+The second parameter is the URL you want to open.
+
+### Launch an Application
+
+```
+launchApplication(String deviceId, String appId)
+```
+
+Opens the application with given appId. To find out what appId constant matches which app, bind the appLauncher channel to a String item and turn the TV to the desired application.
+
+The first parameter is the IP address of your TV. 
+The second parameter is the application id that you want to open.
+
+
+## Full Example
+
+This example assumes the IP of your smart TV is 192.168.2.119.
+
+demo.items:
+
+```
+Switch LG_TV0_Power "TV Power" <television> { channel="lgwebos:WebOSTV:192_168_2_119:power" }
```

Comment:
It is actually not required. Removed it.

Reply:
I though this was to simulate a pushbutton, not on/off switch. This would make sense, as it looks like these controls send commands to the TV on any channel command received. When using a UI like habPanel, this is not a problem, as we can program easily a pushbutton that sends only ON commands. But for other UIs there may be a switch displayed which will toggle between ON and OFF and on each toggle send command to the TV. Adding autoupdate=false will cause the switch to immediately return to the original state after it is pushed to ON, unless binding explicitly updates its state, which will never happen. So it would behave like a pushbutton, which seems to be correct here.

Reply:
autoupdate="false" added